### PR TITLE
Implement Google Login for the Linux app

### LINKED ISSUE #3279: Sign up with Google on Linux
### 💻 Environment
Platform: Linux

### 📒 Description
Add "Sign up with Google" button below the Sign up" button.
To be done after the library part is implemented in #3064.
Obtaining the Google access token is exactly the same as with Google login.

### PR DESCRIPTION
**Pull request by MartinBriza**
### 📒 Description
This PR enables Google Login in the Linux app.
It also fixes some issues in the OAuth code, namely:
 - The dialog no longer closes if you press Enter/Return while writing your email or password (this drove me insane while testing this)
 - The OAuth dialog no longer remembers your account with repeated Login/Signup attempts

### 🕶️ Types of changes
 - **Bug fix** (non-breaking change which fixes an issue)
 - **New feature** (non-breaking change which adds functionality)

### 🤯 List of changes 
 - Linux app now supports Google Login
 - Google Login/Signup dialog on Linux is now more user-friendly

### 👫 Relationships
Closes #3279 #3095

### 🔎 Review hints
Probably similar as with other platforms: Check if Google Login works the same as before and the Google Signup functionality is consistent.

### CODE REVIEW

File: src/ui/linux/TogglDesktop/loginwidget.cpp [hunk 1]
```diff
@@ -46,6 +46,20 @@ void LoginWidget::displayError(
     enableAllControls(true);
 }
 
+void LoginWidget::setGoogleSignup(bool visible) {
+    ui->email->setEnabled(!visible);
+    ui->password->setEnabled(!visible);
+    ui->password->clear();
+    if (visible) {
+        ui->email->setText("You're signing up with your Google Account");
```
Comment (NghiaTranUIT) [inline src/ui/linux/TogglDesktop/loginwidget.cpp]: Look like this LOC add the Text into the Email textbox. Thus, I couldn't select it after failed authentication.

**Comment by MartinBriza** [comment]
The very signing up happens only after you click the `Sign up` button. It felt more natural to let the user log into their google Account first and THEN select their country and agree to the terms. This is intended behavior (except for the part where the "You're signing up.." string stays there even when trying to log in).

File: src/ui/linux/TogglDesktop/loginwidget.cpp (hunk 1 above)
Comment (MartinBriza) [reply src/ui/linux/TogglDesktop/loginwidget.cpp]: This is now fixed by hiding the message when returning to the login screen.

Comment (MartinBriza) [reply src/ui/linux/TogglDesktop/loginwidget.cpp]: I changed the logic to ask for country and agree to the terms first and then enable opening the OAuth dialog. Resolving conversation.

**Comment by NghiaTranUIT** [comment]
Thank for quick address @MartinBriza. Reviewing in second round now 🌮

**Comment by NghiaTranUIT** [comment]
Improvement ticket for the Sign-up with Google at https://github.com/toggl/toggldesktop/issues/3425

**Pull request merged**